### PR TITLE
statistics: replace SimpleDateFormat with DateTimeFormatter

### PR DESCRIPTION
**Pull request by AlexGedeon**
motivation:
use modert java.time API

modification:
SimpleDateFormat was replaced by DateTimeFormatter, Date was replaced by Instant and Calendar by ZonedDateTime

result:
up-to-date code

### CODE REVIEW

**Comment by kofemann** [comment]
ok to test

**Comment by kofemann** [comment]
Hi Alex, the new formats are not fully correct:

	at org.dcache.boot.BootLoader.main(BootLoader.java:125) ~[dcache-core-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
Caused by: java.lang.IllegalArgumentException: Unknown pattern letter: i
	at java.time.format.DateTimeFormatterBuilder.parsePattern(DateTimeFormatterBuilder.java:1661) ~[na:1.8.0_161]
	at java.time.format.DateTimeFormatterBuilder.appendPattern(DateTimeFormatterBuilder.java:1570) ~[na:1.8.0_161]
	at java.time.format.DateTimeFormatter.ofPattern(DateTimeFormatter.java:536) ~[na:1.8.0_161]
	at diskCacheV111.services.PoolStatisticsV0.<clinit>(PoolStatisticsV0.java:160) ~[dcache-core-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	... 15 common frames omitted
2018-03-08 17:57:27


Please have a look at it.

File: modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java [hunk 1]
```diff
@@ -150,90 +157,24 @@
     private static final Escaper AS_PATH_ELEMENT = UrlEscapers.urlPathSegmentEscaper();
 
     private static final String DEFAULT_AUTHOR = "&copy; dCache.org ";
+    private static final DateTimeFormatter _pathFromDate = DateTimeFormatter.ofPattern("yyyy\" + File.separator + \"MM\" + File.separator + \"dd\" + File.separator + \"yyyy-MM-dd-HH'.raw'");
```
Comment (paulmillar) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: Your use of quotes is broken.  The generated String looks like:

    yyyy" + File.separator + "MM" + File.separator[...]

The presence of an 'i' (from "File.separator") prevents dCache from starting:

08 Mar 2018 17:53:39 (System) [] Uncaught exception in thread main
java.lang.ExceptionInInitializerError: null
	at java.lang.Class.forName0(Native Method) ~[na:1.8.0_161]
	at java.lang.Class.forName(Class.java:264) ~[na:1.8.0_161]
	at dmg.cells.nucleus.CellShell$CreateCommand.call(CellShell.java:1358) ~[cells-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at dmg.cells.nucleus.CellShell$CreateCommand.call(CellShell.java:1323) ~[cells-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at org.dcache.util.cli.AnnotatedCommandExecutor.execute(AnnotatedCommandExecutor.java:145) ~[common-cli-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at org.dcache.util.cli.CommandInterpreter$CommandEntry.execute(CommandInterpreter.java:230) ~[common-cli-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at org.dcache.util.cli.CommandInterpreter.doExecute(CommandInterpreter.java:143) ~[common-cli-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at org.dcache.util.cli.CommandInterpreter.command(CommandInterpreter.java:129) ~[common-cli-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at dmg.cells.nucleus.CellShell.objectCommand2(CellShell.java:249) ~[cells-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at dmg.cells.nucleus.CellShell.execute(CellShell.java:2032) ~[cells-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at dmg.cells.nucleus.CellShell.execute(CellShell.java:1992) ~[cells-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at org.dcache.boot.Domain.executeBatchFile(Domain.java:301) ~[dcache-core-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at org.dcache.boot.Domain.executeService(Domain.java:247) ~[dcache-core-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at org.dcache.boot.Domain.start(Domain.java:137) ~[dcache-core-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	at org.dcache.boot.BootLoader.main(BootLoader.java:125) ~[dcache-core-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
Caused by: java.lang.IllegalArgumentException: Unknown pattern letter: i
	at java.time.format.DateTimeFormatterBuilder.parsePattern(DateTimeFormatterBuilder.java:1661) ~[na:1.8.0_161]
	at java.time.format.DateTimeFormatterBuilder.appendPattern(DateTimeFormatterBuilder.java:1570) ~[na:1.8.0_161]
	at java.time.format.DateTimeFormatter.ofPattern(DateTimeFormatter.java:536) ~[na:1.8.0_161]
	at diskCacheV111.services.PoolStatisticsV0.<clinit>(PoolStatisticsV0.java:160) ~[dcache-core-4.2.0-SNAPSHOT.jar:4.2.0-SNAPSHOT]
	... 15 common frames omitted

**Comment by jstarek** [comment]
retest this please

**Comment by kofemann** [comment]
The contributor did not update the patch. Closing.

**Pull request closed without merging**